src/univariate/discrete/negativebinomial2.jl

```diff
@@ -0,0 +1,137 @@
+"""
+Alternative location-scale parameterization of the negative binomial distribution
+in terms of location `μ` and overdispersion `ϕ`.
+
+```math
+P(X = n) = {n + \\phi - 1 \\choose n} (\\frac{\\mu}{\\mu + \\phi})^n (\\frac{\\phi}{\\mu + \\phi})^\\phi, \\quad \\text{for } n = 0,1,2,\\ldots
+```
+
+In terms of the mixture definition, this corresponds to:
+```math
+n \\sim \\text{Poisson}(z), \\quad
+z \\sim \\text{Gamma}(\\phi, \\frac{\\mu}{\\phi})
+```
+This provides a clear interpretation of the negative binomial as an overdispersed
+Poisson with expected location 𝔼[z] = μ and var[z] = μ + μ²/ϕ. It also has advantages in that
+its pmf can be expressed in terms of μ = exp(η), such that it can natively be
+parameterized on the log scale.
+
+```julia
+NegativeBinomial2()        # distribution with μ = 1.0 and ϕ = 1.0
+NegativeBinomial2(μ, ϕ)    # distribution with location μ and overdispersion ϕ
+
+convert(NegativeBinomial{T}, d)         # Parametric conversion to NegativeBinomial
+convert(NegativeBinomial2Log{T}, d)     # Parametric conversion to NegativeBinomial2Log
+convert(NegativeBinomial3{T}, d)        # Parametric conversion to NegativeBinomial3
+```
+
+External links:
+* [Negative binomial distribution (alternative paramerization), Stan](https://mc-stan.org/docs/2_29/functions-reference/nbalt.html)
+Note: the definition above uses the scale parameterization of the Gamma distribution
+(matching Distributions.jl), whereas Stan uses the inverse scale parameterization.
+
+See also: [`NegativeBinomial2Log`](@ref), [`NegativeBinomial3`](@ref)
+"""
+struct NegativeBinomial2{T<:Real} <: DiscreteUnivariateDistribution
+    μ::T
+    ϕ::T
+    NegativeBinomial2{T}(μ::T, ϕ::T) where {T<:Real} = new{T}(μ, ϕ)
+end
+
+function NegativeBinomial2(μ::T, ϕ::T; check_args::Bool=true) where {T<:Real}
+    @check_args NegativeBinomial2 (μ, μ > zero(μ)) (ϕ, ϕ > zero(ϕ))
+    return NegativeBinomial2{T}(μ, ϕ)
+end
+
+NegativeBinomial2(μ::Real, ϕ::Real; check_args::Bool=true) = NegativeBinomial2(promote(μ, ϕ)...; check_args=check_args)
+NegativeBinomial2(μ::Integer, ϕ::Integer; check_args::Bool=true) = NegativeBinomial2(float(μ), float(ϕ); check_args=check_args)
```

Comment:
Is this needed? Integer-valued parameters are fine in principle, but for historical reasons often converted to floats in Distributions.
```suggestion
```

Reply:
This one was a holdover from the extant negativebinomial.jl; I wasn't certain as to why it was there, but thought: might as well carry it forward. Happily removed.